src/app/filter-by-regex.pipe.ts

```diff
@@ -10,21 +10,56 @@ import { LinkData } from './interfaces';
 })
 export class FilterByRegexPipe implements PipeTransform {
 
-  transform(links: LinkData[], regexStr: string,
-      filters: LinkData): LinkData[] {
+  transform(links: LinkData[], filters: any): LinkData[] {
```

Comment:
Try to avoid any here, it's defeat the advantage of using Typescript in the first place.